src/main/kotlin/systems/ajax/malov/stockanalyzer/service/impl/FinnhubStockRecordClientApi.kt

```diff
@@ -15,22 +18,26 @@ class FinnhubStockRecordClientApi(
     @Value("\${api.finnhub.symbols}") private val symbols: List<String>,
 ) : StockRecordClientApi {
 
-    override fun getAllStockRecords(): List<MongoStockRecord> {
+    override fun getAllStockRecords(): Flux<MongoStockRecord> {
         val retrievalDate = Instant.now()
 
-        return symbols.mapNotNull {
-            retrieveStockRecord(it, retrievalDate)
-        }.toList()
+        return Flux.fromIterable(symbols)
+            .flatMap { symbol ->
+                retrieveStockRecord(symbol, retrievalDate)
+            }
+            .mapNotNull { it }
     }
 
     @Suppress("TooGenericExceptionCaught")
-    private fun retrieveStockRecord(symbol: String, retrievalDate: Instant): MongoStockRecord? {
-        try {
-            return finnhubStockApi.quote(symbol).toStockRecord(symbol, retrievalDate)
-        } catch (e: Exception) {
-            log.error("Failed to retrieve data for symbol: $symbol at $retrievalDate. Error: ${e.message}")
-            return null
+    private fun retrieveStockRecord(symbol: String, retrievalDate: Instant): Mono<MongoStockRecord?> {
```

Comment:
I think we don't need nullability any more
```suggestion
    private fun retrieveStockRecord(symbol: String, retrievalDate: Instant): Mono<MongoStockRecord> {
```